core/frontend/src/views/ExtensionManagerView.vue

```diff
@@ -410,61 +431,58 @@ export default Vue.extend({
         })
     },
     async showLogs(extension: InstalledExtensionData) {
-      this.setLoading(extension, true)
-      const ansi = new AnsiUp()
+      this.log_abort_controller = axios.CancelToken.source()
       this.log_output = ''
+      this.log_info_output = `Awaiting logs for ${extension.name}`
+      this.show_log = true
+      let outputBuffer = ''
 
-      back_axios({
-        method: 'get',
-        url: `${API_URL}/log`,
-        params: {
-          container_name: this.getContainerName(extension),
-        },
-        onDownloadProgress: (progressEvent) => {
-          const result = aggregateStreamingResponse(
-            parseStreamingResponse(progressEvent.currentTarget.response),
-            (fragment, buffer) => {
-              // If no logs are available kraken will wait till timeout and stop the stream
-              if (fragment.status === 408) {
-                if (!buffer) {
-                  this.$set(this, 'log_output', ansi.ansi_to_html('No Logs available'))
-                }
-              } else {
-                notifier.pushBackError('EXTENSIONS_LOG_FETCH_FAIL', fragment.error)
-              }
+      const containerName = `extension-${(extension.docker + extension.tag).replace(/[^a-zA-Z0-9]/g, '')}`
+      const fetchLogs = (): void => {
+        const ansi = new AnsiUp()
+        let lastDecode = ''
 
-              /** Only stops if buffer is empty */
-              return Boolean(buffer)
-            },
-          )
+        back_axios({
+          method: 'get',
+          url: `${API_URL}/log`,
+          params: {
+            container_name: containerName,
+          },
+          onDownloadProgress: (progressEvent) => {
+            const result = aggregateStreamingResponse(
+              parseStreamingResponse(progressEvent.currentTarget.response),
+              (_, buffer) => Boolean(buffer),
+            )
 
-          if (result) {
-            this.$set(this, 'log_output', ansi.ansi_to_html(result))
-          }
-          this.show_log = true
-          this.setLoading(extension, false)
-          this.$nextTick(() => {
-            // TODO: find a better way to scroll to bottom
-            const output = document.querySelector(
-              '#app > div.v-dialog__content.v-dialog__content--active > div',
-            ) as HTMLInputElement
-            if (!output) {
+            if (result) {
+              lastDecode = ansi.ansi_to_html(result)
+              this.log_info_output = `Logs for ${extension.name}`
+              this.$set(this, 'log_output', outputBuffer + lastDecode)
+            }
+            this.$nextTick(() => {
+              const logContainer = this.$refs.logContainer as HTMLElement
+              if (logContainer) {
+                logContainer.scrollTop = logContainer.scrollHeight
```

Comment:
Can we avoid scrolling to the bottom every time that a new message is received when the scroll is not in the bottom ?
The use case is that if someone is trying to look at the logs, it should be able to look to previous log messages when new ones are being received.

Reply:
Added a Follow Logs checkbox to allow users to not auto scroll